fl-server/src/handlers.rs

```diff
@@ -268,12 +334,13 @@ pub async fn get_flist_state_handler(
 pub async fn list_flists_handler(State(state): State<Arc<config::AppState>>) -> impl IntoResponse {
     let mut flists: HashMap<String, Vec<FileInfo>> = HashMap::new();
 
-    let rs = visit_dir_one_level(&state.config.flist_dir).await;
+    let rs: Result<Vec<FileInfo>, std::io::Error> =
+        visit_dir_one_level(&state.config.flist_dir, &state).await;
     match rs {
```

Comment:
This feels like too much indentation and it makes it hard to read instead do something like
```
let files = match rs {
   Ok(files) => files,
   Err(err) => {
     return Err(some error);
   }
};

for file in files {
 // process 
}
```